### DISCUSSION TITLE
Rule priority documentation

### DISCUSSION THREAD

**Discussion by oskarjf** [MarimerLLC/csla#3277] [Questions] [answered]
Could you point me in the direction of some documentation of how the priority of rules work? I've been unable to find any.

In v3.8 it was set like this: Csla.Validation.RuleSeverity.Warning
but in 5.5.3 it's an int.

**Comment by danielmartind** [comment, marked as answer]
Priority is to control the order the rules execute, the lower number will run first. You can run a required rule for example before a max length rule.
If you want to use Warning, Info etc you set that in the execute method in the rule calling context. `context.AddWarningResult("Warning message...");
                context.AddInformationResult("Info message...");
                context.AddErrorResult("Error message...");`

**Reply by oskarjf** [reply]
Ofcourse thanks. I was mixing priority with serverity.
